script.js

```diff
@@ -1,12 +1,210 @@
+  const selectShow = document.createElement("select");
+  selectShow.id = _showDDLCtrlId;
+  selectShow.addEventListener("change", function () {
+    refreshEpisodesDDL(selectShow.value);
+  });
+
+  rootElem.appendChild(selectShow);
 }
+function createEPMenu() {
+  const rootElem = document.getElementById("root");
+  const selectMenu = document.createElement("select");
+  selectMenu.id = _epDDLCtrlId;
+  selectMenu.addEventListener("change", function () {
+    drawEpiCard();
+  });
+  rootElem.appendChild(selectMenu);
+}
+function createSearchBar() {
+  const rootElem = document.getElementById("root");
+
+  //search bar
+  const searchBar = document.createElement("input");
+  searchBar.id = "txtSearchBy";
+  searchBar.setAttribute("type", "string");
+  searchBar.addEventListener("input", function () {
+    drawEpiCard();
+  });
+
+  //search result
+  const searchResult = document.createElement("label");
+  searchResult.id = "searchResult";
+
+  // const resultP = document.createElement("p");
+  // resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  // //rootElem.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+
+  // rootElem.insertBefore(resultP, rootElem.firstChild);
+  rootElem.appendChild(searchBar);
+  rootElem.appendChild(searchResult);
+  //drawEpiCard();
+}
+function clearCards() {
+  document
+    .querySelectorAll("#root div.grid-container")
+    .forEach((el) => el.remove());
+}
+function drawEpiCard() {
+  clearCards();
+  //get search words
+  const searchBar = document.getElementById("txtSearchBy");
+  let searchBy = searchBar.value.toLowerCase();
 
-function makePageForEpisodes(episodeList) {
+  //root
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  //selected option
+  const selectMenu = document.getElementById("selName");
+  let selectedName = selectMenu.value;
+
+  //create container
+  const container = document.createElement("div");
+  container.className = "grid-container";
+
+  let found = 0;
+  _localEpisodes.forEach((episode) => {
+    if (
+      //ddl filter
+      (selectedName === "--Please Select--" ||
+        selectedName.includes(episode.name)) &&
+      //search bar filter
+      (!searchBy ||
+        episode.summary.toLowerCase().includes(searchBy) ||
+        episode.name.toLowerCase().includes(searchBy))
+    ) {
+      found++;
+      container.appendChild(getEpiCard(episode));
+    }
+  });
+  rootElem.appendChild(container);
+  const searchResult = document.getElementById("searchResult");
+  if (found > 0 && searchBy) {
+    searchResult.innerText = `Displaying ${found}/${_localEpisodes.length} episodes`;
+  } else {
+    searchResult.innerText = "";
+  }
+}
+function formatVal(val) {
+  return val.toString().padStart(2, "0");
+}
```

Comment:
breaking functions down into useful helper functions like this is really good :)